app/routers/api/files.py

```
@@ -128,38 +132,146 @@ def _serialize_file(
+    is_pinned: Annotated[Optional[bool], FastAPIQuery(description="Filter by pinned status")] = None,
+    min_size: Annotated[Optional[int], FastAPIQuery(description="Minimum file size in bytes")] = None,
+    max_size: Annotated[Optional[int], FastAPIQuery(description="Maximum file size in bytes")] = None,
+    min_mtime: Annotated[Optional[datetime], FastAPIQuery(description="Minimum modification time")] = None,
+    max_mtime: Annotated[Optional[datetime], FastAPIQuery(description="Maximum modification time")] = None,
+    storage_location_id: Annotated[Optional[int], FastAPIQuery(description="Filter by cold storage location ID")] = None,
+    sort_by: Annotated[str, FastAPIQuery(description="Sort field (file_path, file_size, last_seen, storage_type, file_extension)")] = "last_seen",
+    sort_order: Annotated[str, FastAPIQuery(description="Sort order (asc/desc)")] = "desc",
+    page_size: Annotated[int, FastAPIQuery(ge=10, le=500, description="Number of items per page (for pagination)")] = 100,
+    cursor: Annotated[Optional[str], FastAPIQuery(description="Pagination cursor (base64 encoded)")] = None,
+    db: Annotated[Session, Depends(get_db)] = None,
+    current_user: Annotated[User, Depends(get_current_user)] = None,
```

Ruff (0.15.9)

[warning] Unused function argument: `current_user`

(ARG001)